### PR TITLE
travis, build: implement uploading archives to azure

### PR DESCRIPTION
**Pull request by karalabe**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,14 +32,23 @@ matrix:
...
+        - go run build/ci.go archive -type tar -signer LINUX_SIGNING_KEY -upload gethstore/builds
+
+    # This builder does the OSX Azure uploads
+    - os: osx
+      go: 1.7
+      env:
+        - azure-osx-amd64
+      script:
+        - go run build/ci.go install
+        - go run build/ci.go archive -type zip -signer OSX_SIGNING_KEY -upload gethstore/builds
```
Comment (fjl) [inline .travis.yml]: Can we do the upload from the main osx builder?

File: build/ci.go [hunk 2]
```diff
@@ -262,6 +266,33 @@ func doArchive(cmdline []string) {
...
+			}
+			if err := azureUpload(archive, archive, auth); err != nil {
+				log.Fatal(err)
+			}
+			if *signer != "" {
+				if err := azureUpload(archive+".asc", archive+".asc", auth); err != nil {
+					log.Fatal(err)
+				}
+			}
+		}
```
Comment (fjl) [inline build/ci.go]: Please put this into a function.

File: build/ci.go [hunk 3]
```diff
@@ -465,6 +496,63 @@ func stageDebianSource(tmpdir string, meta debMetadata) (pkgdir string) {
...
+	defer in.Close()
+
+	out, err := os.Create(output)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	return openpgp.ArmoredDetachSignText(out, keys[0], in, nil)
+}
```
Comment (fjl) [inline build/ci.go]: Please put these two functions into the internal/build package. They are kind of standalone.

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 47420ce3633b4c8576738d56c618c89b2ac47b98
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Pull request merged**